probe-rs/src/vendor/st/mod.rs

```diff
@@ -42,7 +42,12 @@ impl Vendor for St {
         {
             DebugSequence::Arm(Stm32Armv7::create())
         } else if chip.name.starts_with("STM32H7") {
-            DebugSequence::Arm(Stm32h7::create())
+            let ap = if chip.name.starts_with("STM32H7S") || chip.name.starts_with("STM32H7R") {
+                1
+            } else {
+                2
+            };
```

Comment:
Would you move this logic into `Stm32h7::create` instead? I think it's better if the sequence contained everything it needs.

Reply:
OK, but I still have to pass in a parameter for which SoC line.  I made it look like `Stm32Armv6::create()`; is that what you wanted?

Reply:
Yup, thanks. If you could please resolve the rustdoc issues, we can get this merged.